sktime/forecasting/ets.py

```diff
@@ -0,0 +1,338 @@
+
+            # Fit model, adapted from:
+            # https://github.com/robjhyndman/forecast/blob/master/R/ets.R
+
+            # Initialise iterator
+            def _iter(error_range, trend_range, seasonal_range, damped_range):
+                for error, trend, seasonal, damped in product(error_range,
+                                                              trend_range,
+                                                              seasonal_range,
+                                                              damped_range):
+                    if trend is None and damped:
+                        continue
+
+                    if self.restrict:
+                        if error == 'add' and (trend == 'mul' or
+                                               seasonal == 'mul'):
+                            continue
+                        if error == 'mul' and trend == 'mul' and \
+                                seasonal == 'add':
+                            continue
+                        if self.additive_only and (error == 'mul' or
+                                                   trend == 'mul' or
+                                                   seasonal == 'mul'):
+                            continue
+
+                    yield error, trend, seasonal, damped
+
+            # Fit function
+            def _fit(error, trend, seasonal, damped):
+                _forecaster = _ETSModel(
+                    y,
+                    error=error,
+                    trend=trend,
+                    damped_trend=damped,
+                    seasonal=seasonal,
+                    seasonal_periods=self.sp,
+                    initialization_method=self.
+                    initialization_method,
+                    initial_level=self.initial_level,
+                    initial_trend=self.initial_trend,
+                    initial_seasonal=self.initial_seasonal,
+                    bounds=self.bounds,
+                    dates=self.dates,
+                    freq=self.freq,
+                    missing=self.missing
+                )
+                _fitted_forecaster = _forecaster.fit(
+                    start_params=self.start_params,
+                    maxiter=self.maxiter,
+                    full_output=self.full_output,
+                    disp=self.disp,
+                    callback=self.callback,
+                    return_params=self.return_params
+                )
+                return _forecaster, _fitted_forecaster
+
+            # Fit models
+            _fitted_results = Parallel(n_jobs=self.n_jobs)(
+                delayed(_fit)(error, trend, seasonal, damped)
+                for error, trend, seasonal, damped in _iter(
+                        error_range, trend_range, seasonal_range, damped_range
+                    ))
+
+            # Select best model based on information criterion
+            for result in _fitted_results:
```

Comment:
We can do this in a loop, but I wonder if there's a faster way using `np.arrray` where we can simply call something like `np.argmax(_ics)` to find the index of the model with the highest IC, but it's always a bit tricky to get the returned list (`_fitted_results`) into an array ...